src/hist/intervals.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, Optional
+
+import numpy as np
+from scipy import stats
+
+from .typing import Literal
+
+__all__ = ("poisson_interval", "clopper_pearson_interval", "ratio_uncertainty")
+
+
+def poisson_interval(
+    values: np.ndarray, variances: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    r"""
+    The Frequentist coverage interval for Poisson-distributed observations.
+
+    What is calculated is the "Garwood" interval,
+    c.f. https://www.ine.pt/revstat/pdf/rs120203.pdf or
+    http://ms.mcmaster.ca/peter/s743/poissonalpha.html.
+    For weighted data, this approximates the observed count by
+    ``values**2/variances``, which effectively scales the unweighted
+    Poisson interval by the average weight.
+    This may not be the optimal solution: see https://arxiv.org/abs/1309.1287
+    for a proper treatment.
+
+    When a bin is zero, the scale of the nearest nonzero bin is substituted to
+    scale the nominal upper bound.
+    If all bins zero, a warning is generated and interval is set to ``values``.
+
+    Args:
+        values: Sum of weights.
+        variances: Sum of weights squared.
+        coverage: Central coverage interval.
+          Default is one standard deviation, which is roughly ``0.68``.
+
+    Returns:
+        The Poisson central coverage interval.
+    """
+    # Parts originally contributed to coffea
+    # https://github.com/CoffeaTeam/coffea/blob/8c58807e199a7694bf15e3803dbaf706d34bbfa0/LICENSE
```

Comment:
@lgray @nsmith- @eduardo-rodrigues moving discussion here now.

Reply:
@nsmith-, I'd like at least your sign-off on this, just to be safe. :)

Reply:
To be clear, I was just adding this as I thought it was required under BSD3, not because I wanted to make things complicated. So as long as @henryiii and the `coffea` team are happy with whatever is here or not here, I'm good.

Reply:
The attribution looks OK to me